Allow Assigning Generic Hooks To Type `Hook`

# Suggestion (TS) - Allow Assigning Generic Hooks To Type `Hook`

## Summary
Phoenix LiveView's TypeScript definitions prevent the use of generic `Hook<T>` types when the generic parameter `T` has required properties. The type system incorrectly assumes the generic parameter should be assignable to `object` (which is effectively `{}`), causing compilation errors when using custom interfaces with required properties.

## Use Case
Developers want to create Phoenix LiveView hooks with strongly-typed custom interfaces to improve type safety.

## Expected Behavior
The following code should compile without errors:

```typescript
import { Hook, HooksOptions } from "phoenix_live_view";

interface LinksInTab {
  x: string;
}

const LinksInTab: Hook<LinksInTab> = {
  x: "x"
};

export const Hooks: HooksOptions = {
  LinksInTab  // Should work
};
```

## Actual Behavior
TypeScript compiler throws the following error:

```
index.ts(12,3): error TS2322: Type 'Hook<LinksInTab>' is not assignable to type 'typeof ViewHook | Hook<object>'.
  Type 'Hook<LinksInTab>' is not assignable to type 'Hook<object>'.
    Property 'x' is missing in type '{}' but required in type 'LinksInTab'.
```
According to my understanding, the issue appears to stem from a type constraint mismatch. My suspicion is that the `HooksOptions` type definition expects `Hook<object>` (essentially `Hook<{}>`), but TypeScript cannot safely assign the more specific `Hook<LinksInTab>` type to this constraint since `LinksInTab` has required properties that `object` doesn't guarantee.

## Reproduction
A minimal reproduction case is available at: [https://github.com/IdoLeshkowitz/reproduce-live-view-hook-unassignable-generic-type]

### Steps to reproduce:
1. Clone the repository
2. Run `npm install` 
3. Run `npm run compile`
4. Observe the TypeScript compilation error

### Environment:
- TypeScript: 5.9.2
- phoenix_live_view: 1.1.7

I will be glad to suggest a fix, just wanna confirm that this is not intended for some reason.

--- a/assets/js/phoenix_live_view/view_hook.ts
+++ b/assets/js/phoenix_live_view/view_hook.ts
@@ -159,7 +159,7 @@ export interface HookInterface {
 
 // based on https://github.com/DefinitelyTyped/DefinitelyTyped/blob/fac1aa75acdddbf4f1a95e98ee2297b54ce4b4c9/types/phoenix_live_view/hooks.d.ts#L26
 // licensed under MIT
-export interface Hook<T = object> {
+export interface Hook<T = never> {
   /**
    * The mounted callback.
    *

--- a/assets/js/phoenix_live_view/view_hook.ts
+++ b/assets/js/phoenix_live_view/view_hook.ts
@@ -159,7 +159,7 @@ export interface HookInterface {
 
 // based on https://github.com/DefinitelyTyped/DefinitelyTyped/blob/fac1aa75acdddbf4f1a95e98ee2297b54ce4b4c9/types/phoenix_live_view/hooks.d.ts#L26
 // licensed under MIT
-export interface Hook<T = never> {
+export interface Hook<out T = object> {
   /**
    * The mounted callback.
    *

